app/assets/javascripts/discourse/app/components/composer-user-selector.js

```diff
@@ -17,78 +16,40 @@ export default Component.extend({
     }
   },
 
-  @observes("usernames")
-  _checkWidth() {
-    let width = 0;
-    const $acWrap = $(this.element).find(".ac-wrap");
-    const limit = $acWrap.width();
-    this.set("defaultUsernameCount", 0);
-
-    $acWrap
-      .find(".item")
-      .toArray()
-      .forEach((item) => {
-        width += $(item).outerWidth(true);
-        const result = width < limit;
-
-        if (result) {
-          this.incrementProperty("defaultUsernameCount");
-        }
-        return result;
-      });
-
-    if (width >= limit) {
-      this.set("shouldHide", true);
-    } else {
-      this.set("shouldHide", false);
-    }
-  },
-
-  @observes("shouldHide")
-  _setFocus() {
-    const selector =
-      "#reply-control #reply-title, #reply-control .d-editor-input";
-
-    if (this.shouldHide) {
-      $(selector).on("focus.composer-user-selector", () => {
-        this.set("showSelector", false);
-        this.appEvents.trigger("composer:resize");
-      });
-    } else {
-      $(selector).off("focus.composer-user-selector");
-    }
-  },
-
-  @discourseComputed("usernames")
-  splitUsernames(usernames) {
-    return usernames.split(",");
+  @discourseComputed("recipients")
+  splitRecipients(recipients) {
+    return recipients ? recipients.split(",").filter(Boolean) : [];
   },
 
-  @discourseComputed("splitUsernames", "defaultUsernameCount")
-  limitedUsernames(splitUsernames, count) {
-    return splitUsernames.slice(0, count).join(", ");
-  },
-
-  @discourseComputed("splitUsernames", "defaultUsernameCount")
-  hiddenUsersCount(splitUsernames, count) {
-    return `${splitUsernames.length - count} ${I18n.t("more")}`;
+  __updateGroups(selected, newGroups) {
+    const groups = [];
+    this.__groups.forEach((existing) => {
+      if (selected.includes(existing)) {
+        groups.addObject(existing);
+      }
+    });
+    newGroups.forEach((newGroup) => {
+      if (!groups.includes(newGroup)) {
+        groups.addObject(newGroup);
+      }
+    });
+    this.setProperties({
+      __groups: groups,
```

Comment:
cant we find a better name for these groups which are not groups ?

Reply:
Sorry I'm not sure I understand what you mean here? I thought the underscore at the beginning is a JS convention that denotes the instance variable is 'private'?

Reply:
one yes, not two :p

Reply:
Discussed this internally with Joffrey and decided to tackle it later.